app/src/main/java/com/infomaniak/drive/data/services/UploadWorker.kt

```diff
@@ -71,47 +72,30 @@ class UploadWorker(appContext: Context, params: WorkerParameters) : CoroutineWor
     private var uploadedCount = 0
 
     override suspend fun doWork(): Result {
+
+        Log.d(TAG, "UploadWorker starts job!")
         contentResolver = applicationContext.contentResolver
-        Log.d(TAG, "UploadWorker start job !")
 
         // Checks if the maximum number of retry allowed is reached
-        if (runAttemptCount >= MAX_RETRY_COUNT) {
-            return Result.failure()
-        }
+        if (runAttemptCount >= MAX_RETRY_COUNT) return Result.failure()
 
-        // Move the service to foreground
-        applicationContext.uploadServiceNotification().apply {
-            setContentTitle(applicationContext.getString(R.string.notificationUploadServiceChannelName))
-            setForeground(ForegroundInfo(NotificationUtils.UPLOAD_SERVICE_ID, build()))
-        }
+        moveServiceToForeground()
 
         return try {
+
             // Check if we have the required permissions before continuing
-            if (!applicationContext.hasPermissions(DrivePermissions.permissions)) {
-                UploadNotifications.permissionErrorNotification(applicationContext)
-                Log.d(TAG, "UploadWorker no permissions")
-                return Result.failure()
-            }
+            checkPermissions()?.let { return it }
 
-            // Retrieve the latest media that have not been taken in the sync
-            val appSyncSettings = UploadFile.getAppSyncSettings()
-            appSyncSettings?.let {
-                Log.d(TAG, "UploadWorker check locals")
-                checkLocalLastMedias(it)
-            }
+            // Retrieve the latest media that have not been synced
+            val appSyncSettings = retrieveLatestNotSyncedMedia()
 
-            // Check if the user has cancelled the upload and no files to sync
-            val isCancelledByUser = inputData.getBoolean(CANCELLED_BY_USER, false)
-            if (UploadFile.getAllPendingUploadsCount() == 0 && isCancelledByUser) {
-                UploadNotifications.showCancelledByUserNotification(applicationContext)
-                Log.d(TAG, "UploadWorker cancelled by user")
-                return Result.success()
-            }
+            // Check if the user has cancelled the uploads and there is no more files to sync
+            checkRemainingUploadsAndCancellationStatus()?.let { return it }
```

Comment:
This name is misleading, here it's just a user cancellation check, `CancellationStatus is too generic`.